crates/red_knot_python_semantic/src/types.rs

```diff
@@ -636,20 +626,29 @@ impl<'db> UnionType<'db> {
         self.elements(db).contains(&ty)
     }
 
-    /// Apply a transformation function to all elements of the union,
-    /// and create a new union from the resulting set of types
-    pub fn map(
-        &self,
+    /// Create a union from a list of elements
+    /// (which may be eagerly simplified into a different variant of [`Type`] altogether)
+    pub fn from_elements(
```

Comment:
Nit: I would probably call the method `from_iter`

Reply:
I prefer `from_elements()` :( but it looks like the prevailing opinion may be against me here...

Reply:
Unfortunately I don't have a strong opinion here to settle this debate :) Both names seem fine to me. I guess I will also come down slightly on the side of `from_iter`, because I think it is obvious that the iterable would have to be an iterable of union elements (what else could it be?), whereas it's less obvious that `from_elements` would take an iterable (though tbh I think this is pretty obvious too, thus my lack of strong opinion...)

I would be curious what the reason is for @MichaReiser and @dhruvmanila preference.